Src/FluentAssertions/Formatting/LineCollection.cs

```diff
@@ -0,0 +1,97 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using FluentAssertions.Execution;
+
+namespace FluentAssertions.Formatting;
+
+/// <summary>
+/// A collection of lines that will throw a <see cref="MaxLinesExceededException"/> when the number of lines
+/// exceeds the maximum.
+/// </summary>
+internal class LineCollection(int maxLines) : IEnumerable<Line>
+{
+    private readonly List<Line> lines = new();
+
+    public int Count => lines.Count;
+
+    public bool HasLinesBeyond(Line line)
+    {
+        // Null means that we're referring to the top of the list
+        return (line is null && lines.Count > 1) || (line is not null && lines.IndexOf(line) < (lines.Count - 1));
+    }
+
+    public void Add(Line line)
+    {
+        lines.Add(line);
+        OnCollectionIsModified();
+    }
+
+    public void AddLineAfter(Line line, Line newLine)
+    {
+        int index = lines.IndexOf(line);
+
+        Insert(index + 1, newLine);
+    }
+
+    public void InsertLineBefore(Line line, Line newLine)
+    {
+        int index = lines.IndexOf(line);
+
+        Insert(Math.Max(index, 0), newLine);
```

Comment:
It seems `index` is -1 when we `line` is `null` in `Anchor.InsertLineOrFragment`.

Would it make sense to instead
```
if (line is null)
{
    // not sure what to call it.
    parent.AddFirstLine(fragmentWithWhitespace);
}
else if (lineWasEmptyAtCreation)
{
    parent.InsertLineBefore(line, fragmentWithWhitespace);
}
else
{
    parent.AddLineAfter(line, fragmentWithWhitespace);
}
```

Alternatively to explicitly accept a null `line`.
```cs
int index = line is not null ? lines.IndexOf(line) : 0;
```

Reply:
The mutation testing showed that we can set `index` to `0` without breaking any tests.
This means that `line` is always `null`.
I verified this by breaking the debugger if `line` was `null` and this wasn't hit when running tests.

Reply:
Indeed. analysis.